### ISSUE TITLE
[Blenderbot] Does Training data have to be diffrent for a 3B model finetuning?

### ISSUE THREAD

**Issue by fral1013** [facebookresearch/ParlAI#3343]
Hello, while i was trying to finetune the diffrent Blenderbot models i was running into the issue that i could train the 90M just fine with a result that can talk back to me with the trained data, but when i try a similiar thing with the same training data on the 3B model i only get gibberish as answers. So my general question is if i have to feed the model some diffrent kind of data if i want to finetune the 3B model.

Here are some information on how i did stuff so far:

Training command for the 90M model: (which is working fine and the outcome model works)
`
python multiprocessing_train -t blended_skill_talk,fromfile:parlaiformat --fromfile_datapath "/home/blender/testmodel/data/output.txt" -m transformer/generator --multitask-weights 1,1 --init-model zoo:tutorial_transformer_generator/model --dict-file zoo:tutorial_transformer_generator/model.dict --embedding-size 512 --n-layers 8 --ffn-size 2048 --dropout 0.1 --n-heads 16 --learn-positional-embeddings True --n-positions 512 --variant xlm --activation gelu --skip-generation True --fp16 True --text-truncate 512 --label-truncate 128 --dict-tokenizer bpe --dict-lower True -lr 1e-06 --optimizer adamax --lr-scheduler reduceonplateau --gradient-clip 0.1 --betas 0.9,0.999 --update-freq 1 --log-every-n-secs 30 --attention-dropout 0.0 --relu-dropout 0.0 --skip-generation True --batchsize 64 -veps 10 -vp 15 -stim 2700 -vme 20000 -bs 60 -vmt ppl -vmm min --save-after-valid True --model-file "/home/blender/testmodel/90Mmodel/test_train_90M"
`

Training command for the 3B model:
`
python train_model -t blended_skill_talk,fromfile:parlaiformat --fromfile_datapath "/home/blender/testmodel/data/output.txt" --multitask-weights 1,1 -veps 1.2 --attention-dropout 0.0 --batchsize 16 --model transformer/generator --embedding-size 2560 --ffn-size 10240 --variant prelayernorm --n-heads 32 --n-positions 128 --n-encoder-layers 2 --n-decoder-layers 24 --history-add-global-end-token end --delimiter '  ' --dict-tokenizer bpe  --dropout 0.1 --fp16 True --init-model zoo:blender/blender_3B/model --dict-file zoo:blender/blender_3B/model.dict --label-truncate 128 --log_every_n_secs 30 -lr 7e-06 --lr-scheduler reduceonplateau --lr-scheduler-patience 3 --optimizer adam --relu-dropout 0.0 --activation gelu --model-parallel true --save-after-valid True --text-truncate 128 --truncate 128 --warmup_updates 100 --fp16-impl mem_efficient --update-freq 2 --gradient-clip 0.1 --skip-generation True -vp 10 -vmt ppl --model-file "/home/blender/testmodel/3Bmodel/test_train_3B"
`

Outcome of the trained 3B model (landing at a PPL of around 1,04)
![3BModeloutcome](https://user-images.githubusercontent.com/56153613/103639286-6e35e000-4f4e-11eb-8b46-76b5420e503b.PNG)

I noticed that the diffrent models use diffrent types of dictionaries (one being full words with a certain "amount" next to it and the other being sometimes just parts of words paired with a simple 1) so my question is that it maybe has to do something with the used training data and that maybe it has to be diffrent for the 2 models tested.

Here is 1 epoch of the training data that has been used (file is called "output.txt", that was used in both trainings)

"text:Do you know the person Andreas Brehmer?		labels:Yes, I know him.		
text:Do you know the nationality of the person?		labels:Yes, he is german.		
text:What do you know when the person was born?		labels:He was born in the year 1964		
text:What do you know if the person passed away?		labels:He passed away in the country Germany.
episode_done:True"
(this goes on for multiple persons)

The training part itself works just fine so if i would have to guess it must be either the quality of the data or the form of the data.
I hope that i provided enough data so that someone could help my with my questions.

**Comment by stephenroller** [comment]
The tokenizer argument for the 3B is wrong. It should be --dict-tokenizer bytelevelbpe.

Double check all the arguments for fine tuning the 3B on our website.

**Comment by fral1013** [comment]
Oh my.. it really was that simple, cant believe that i missed that.

I changed it and started a new training session and now the bot answers with actual words again thanks alot!

**Comment by stephenroller** [comment]
FYI there's some newer distilled BB models that are like 400M and 1B. A bit easier to work with.

**Comment by Hilal-Urun** [comment]
Hi, I want to ask something about finetuning. For example we want to finetune with wizard of wikipedia dialogue task.  After process, program will know current informations, or we have to create our own dataset

**Comment by stephenroller** [comment]
The model already "knows" a lot from its pretraining (it also knows a lot of wrong stuff. The internet isn't reliable). Fine-tuning on WoW mostly doesn't teach it new facts, but really turns it into an insufferable know-it-all. It may learn some new things, but my instinct is that it mostly "exposes" what it already knows.

**Comment by PolKul** [comment]
hello, I would like to fine-tune the distilled 1B model. You have a command for the full 3B model as follows:
`parlai train_model -t blended_skill_talk,wizard_of_wikipedia,convai2:normalized,empathetic_dialogues --multitask-weights 1,3,3,3 -veps 0.25 --attention-dropout 0.0 --batchsize 128 --model transformer/generator --embedding-size 2560 --ffn-size 10240 --variant prelayernorm --n-heads 32 --n-positions 128 --n-encoder-layers 2 --n-decoder-layers 24 --history-add-global-end-token end --delimiter '  ' --dict-tokenizer bytelevelbpe  --dropout 0.1 --fp16 True --init-model zoo:blender/reddit_3B/model --dict-file zoo:blender/reddit_3B/model.dict --label-truncate 128 --log_every_n_secs 10 -lr 7e-06 --lr-scheduler reduceonplateau --lr-scheduler-patience 3 --optimizer adam --relu-dropout 0.0 --activation gelu --model-parallel true --save-after-valid True --text-truncate 128 --truncate 128 --warmup_updates 100 --fp16-impl mem_efficient --update-freq 2 --gradient-clip 0.1 --skip-generation True -vp 10 -vmt ppl -vmm min --model-file /tmp/test_train_27B`

Can you please give me advice how to modify that command so that it could work for 1B distilled model?

**Comment by farnazgh** [comment]
> The model already "knows" a lot from its pretraining (it also knows a lot of wrong stuff. The internet isn't reliable). Fine-tuning on WoW mostly doesn't teach it new facts, but really turns it into an insufferable know-it-all. It may learn some new things, but my instinct is that it mostly "exposes" what it already knows.

As the current model has been trained on a huge data, do you think fine-tuning on which amount of data (approximately) can change the attitude of the bot (considering 90M version)? the same question for adopting the language of Blenderbot to other non-English languages?
